corehq/blobs/s3db.py

```diff
@@ -162,15 +162,33 @@ def _s3_bucket(self, create=False):
 
 
 def is_not_found(err, not_found_codes=["NoSuchKey", "NoSuchBucket", "404"]):
-    return (err.response["Error"]["Code"] in not_found_codes or
-        err.response.get("Errors", {}).get("Error", {}).get("Code") in not_found_codes)
+    if hasattr(err, 'response'):
+        # err quacks like a ClientError
+        return (
+            err.response["Error"]["Code"] in not_found_codes
+            or err.response.get("Errors", {}).get("Error", {}).get("Code") in not_found_codes
+        )
+
+    # err might be an SSLError. It seems unintuitive to get an SSLError
+    # when a blob is not found. And yet ...
+    #
+    #     botocore.exceptions.SSLError: \
+    #     SSL validation failed for https://s3.amazonaws.com/...blobdb/abc123 \
+    #     [Errno 2] No such file or directory
+    #
+    # ... which comes from ...
+    #
+    #     Traceback (most recent call last):
+    #       File ".../urllib3/util/ssl_.py", line 336, in ssl_wrap_socket
+    #     FileNotFoundError: [Errno 2] No such file or directory
+    return "No such file or directory" in str(err)
```

Comment:
I'm not sure whether checking for a string in an error is the best approach. An alternative might be to inspect the traceback, but that also feels ... awkward. Curious what you think.

Reply:
What file is not found? Is it the actual file being requested, or some file related to SSL auth? This doesn't seem right to me. Is this error coming from Amazon S3 or some other blob db backend?

Reply:
I think it's the actual file. I'll email you the full traceback.

Reply:
After reviewing a stack trace shared via email, I think this is actually an error in the SSL handshake, and not caused by a missing S3 blob. The error is originating in [urllib3/util/ssl_.py](https://github.com/urllib3/urllib3/blob/1.25.6/src/urllib3/util/ssl_.py#L334-L343) where it's loading CA cert files:
```py
    if ca_certs or ca_cert_dir:
        try:
            context.load_verify_locations(ca_certs, ca_cert_dir)
        except IOError as e:  # Platform-specific: Python 2.7
            raise SSLError(e)
```
I'm guessing this is caused by a bad client-side SSL configuration and not a result of missing blobs.

Reply:
Thank you @millerdev ! I should have checked urllib3. I'll close this PR.

Reply:
Adding a bit more context for posterity: the error was caused by an HQ deploy, which mutated the virtualenv and/or deleted the previous release. These mutations caused files that urllib3 SSL verification depends on to become unavailable while a process initiated from the previous release (and in which the error occurred) was still running.